src/block_cache/block_chunkcache.c

```diff
@@ -55,7 +57,66 @@ __chunkcache_bitmap_find_free(WT_SESSION_IMPL *session, size_t *bit_index)
             *bit_index = ((bitmap_size * 8) + j);
             return (0);
         }
-    WT_RET_MSG(session, ENOMEM, "Bitmap has no free chunks as cache is full.");
+    return (ENOSPC);
+}
+
+/*
+ * __chunkcache_bitmap_alloc --
+ *     Find the bit index to allocate.
+ */
+static int
+__chunkcache_bitmap_alloc(WT_SESSION_IMPL *session, size_t *bit_index)
+{
+    WT_CHUNKCACHE *chunkcache;
+    size_t num_chunks;
+    uint8_t *map_byte_p, map_byte_expected, map_byte_mask;
+
+    chunkcache = &S2C(session)->chunkcache;
+    num_chunks = chunkcache->capacity / chunkcache->chunk_size;
+
+retry:
+    /* Use the bitmap to find a free slot for a chunk in the cache. */
+    WT_RET(__chunkcache_bitmap_find_free(session, bit_index));
+
+    /* Bit index should be less than the maximum number of chunks that can be allocated. */
+    WT_ASSERT(session, *bit_index < num_chunks);
+
+    /* Mark the free chunk in the bitmap as in use. */
+    map_byte_p = &chunkcache->free_bitmap[*bit_index / 8];
+
+    /*
+     * Ensure to prevent multiple reads by typecasting to volatile. FIXME WT-11285 Use the
+     * WT_READ_ONCE macro instead.
+     */
+    map_byte_expected = (*(volatile uint8_t *)map_byte_p);
```

Comment:
@ajmorton, currently this helps prevent multiple reads of the variable. Fix me added, `WT_READ_ONCE` until WT-11285 is resolved.

Reply:
Looks good. Thanks!